lock-api/src/test/java/com/palantir/lock/client/AsyncTimeLockUnlockerTest.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.client;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.awaitility.Awaitility;
+import org.awaitility.Duration;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import com.google.common.util.concurrent.Uninterruptibles;
+import com.palantir.common.concurrent.PTExecutors;
+import com.palantir.lock.v2.LockToken;
+import com.palantir.lock.v2.TimelockService;
+
+public class AsyncTimeLockUnlockerTest {
+    private static final int ONE_THOUSAND = 1000;
+
+    private List<LockToken> tokenList;
+    private List<LockToken> unlockedTokens;
+
+    private TimelockService timelockService = mock(TimelockService.class);
+    private AtomicReference<Set<LockToken>> outstandingTokens = new AtomicReference<>(Sets.newConcurrentHashSet());
```

Comment:
The real impl starts out with an `ImmutableSet` here - why change the behavior for the test?

Reply:
No reason to - and yeah this should have been an `ImmutableSet` in the previous version. It's now a `ConcurrentHashSet` as that's what the real impl uses

Reply:
Actually removed as it's no longer meaningful to test that adding directly to the set won't leave stragglers (it will, but that's not a valid state any longer, at least in theory)